druid/src/core.rs

```diff
@@ -423,15 +423,25 @@ impl<T: Data, W: Widget<T>> WidgetPod<T, W> {
             LifeCycle::HotChanged(_) => false,
             LifeCycle::RouteFocusChanged { old, new } => {
                 self.state.request_focus = None;
-                let this_changed = old.map(|_| false).or_else(|| new.map(|_| true));
+
+                let this_changed = if *old == Some(self.state.id) {
+                    Some(false)
+                } else if *new == Some(self.state.id) {
+                    Some(true)
+                } else {
+                    None
+                };
+
                 if let Some(change) = this_changed {
                     let event = LifeCycle::FocusChanged(change);
                     self.inner.lifecycle(ctx, &event, data, env);
                     false
                 } else {
-                    //FIXME: I think this is a bug? we should be checking `self.state.children`
-                    old.map(|id| ctx.children.contains(&id)).unwrap_or(false)
-                        || new.map(|id| ctx.children.contains(&id)).unwrap_or(false)
+                    old.map(|id| self.state.children.contains(&id))
+                        .unwrap_or(false)
+                        || new
```

Comment:
Minor style nit: The or operator here tripped me up ("Why is there a closure here?!")
Can we spell this
`[...]
.unwrap_or(false)
||
new.map(...)
[...]`
Or will rustfmt move it back this way?

Reply:
ah interesting, I hadn't thought to see that as a closure. This formatting is rustfmt's doing, but I can rewrite this using `or_else`, which should be clearer.

Reply:
Thanks. I only just now realized that GH discarded my newlines in what I'd written. Edited above to what I intended.